src/box/iproto.cc

```diff
@@ -1256,10 +1259,15 @@ iproto_connection_on_input(ev_loop *loop, struct ev_io *watcher,
 			return;
 		}
 		/* Read input. */
-		int nrd = sio_read(fd, in->wpos, ibuf_unused(in));
+		ssize_t nrd = iostream_read(io, in->wpos, ibuf_unused(in));
 		if (nrd < 0) {                  /* Socket is not ready. */
-			if (! sio_wouldblock(errno))
+			if (nrd == IOSTREAM_ERROR)
 				diag_raise();
+			int events = iostream_status_to_events(nrd);
+			if (con->input.events != events) {
+				ev_io_stop(loop, &con->input);
+				ev_io_set(&con->input, con->input.fd, events);
+			}
```

Comment:
https://github.com/tarantool/tarantool/blob/5e003c5a64094273dd2a9769792b48f08b0a4ccd/third_party/libev/ev.h#L326-L332

https://github.com/tarantool/tarantool/blob/5e003c5a64094273dd2a9769792b48f08b0a4ccd/third_party/libev/ev.h#L267-L282

`ev_io_modify()` won't work, because it simply updates `events` without stopping the watcher:

https://github.com/tarantool/tarantool/blob/5e003c5a64094273dd2a9769792b48f08b0a4ccd/third_party/libev/ev.h#L706